grayskull/pypi/pypi.py

```diff
@@ -787,15 +809,23 @@ def _extract_requirements(self, metadata: dict) -> dict:
         :return: all requirement section
         """
         name = metadata["name"]
-        requires_dist = PyPi._format_dependencies(metadata.get("requires_dist"), name)
-        setup_requires = (
-            metadata.get("setup_requires") if metadata.get("setup_requires") else []
-        )
-        host_req = PyPi._format_dependencies(setup_requires, name)
+        if metadata.get("requires_dist"):
+            requires_dist = PyPi._format_dependencies(metadata.get("requires_dist"), name)
+        else:
+            requires_dist = metadata.get("requires_dist")
+
+        if metadata.get("setup_requires"):
+            setup_requires = (
+                metadata.get("setup_requires") if metadata.get("setup_requires") else []
+            )
+            host_req = PyPi._format_dependencies(setup_requires, name)
+        else:
+            host_req = metadata.get("setup_requires")
 
         if not requires_dist and not host_req and not metadata.get("requires_python"):
             return {"host": sorted(["python", "pip"]), "run": ["python"]}
```

Comment:
Why the `sorted` here? I don't think we need to ensure order here and anyway it's two elements we can sort ourselves.

Reply:
Okay. It was there in the code already. And yes, I guess it's not needed.